lib/ES/Util.pm

```diff
@@ -253,21 +253,40 @@ sub build_single {
 }
 
 #===================================
-# Emulates how we've been using docbook's -docinfo.xml feature by allowing us
-# to add our own extra html to the title page of books.
+# Customize a book's title page.
+#
+# By default, the title page shows just the table of contents.
+#
+# If a file exists named `<index>-custom-title-page.html`, the table of contents
+# will be replaced by the contents of this file.  Otherwise, if a file named
+# `<index>-extra-title-page.html` exists, its contents will be added above the
+# table of contents. (Note that the `<index>` may be different based on the
+# book's "index" file.
 #===================================
-sub _add_extra_title_page {
+sub _customize_title_page {
 #===================================
     my ( $index, $html_file ) = @_;
-    my $extra_title_page = $index->basename;
-    $extra_title_page =~ s/(\.x)?\.a(scii)?doc$/-extra-title-page.html/ || die;
+    my $index_basename = $index->basename;
+
+    (my $custom_title_page = $index_basename) =~ s/(\.x)?\.a(scii)?doc$/-custom-title-page.html/ || die;
+    $custom_title_page = $index->parent->file( $custom_title_page );
+
+    (my $extra_title_page = $index_basename) =~ s/(\.x)?\.a(scii)?doc$/-extra-title-page.html/ || die;
     $extra_title_page = $index->parent->file( $extra_title_page );
-    if ( -e $extra_title_page ) {
-        $extra_title_page = $extra_title_page->slurp( iomode => '<:encoding(UTF-8)' );
+
+    if ( -e $custom_title_page ) {
+        my $custom_contents = $custom_title_page->slurp( iomode => '<:encoding(UTF-8)' );
+        my $contents = $html_file->slurp( iomode => '<:encoding(UTF-8)' );
+        $contents =~ s|<!--START_TOC-->.*<!--END_TOC-->|$custom_contents|sm or
+            die "Couldn't set custom contents in file";
+        $html_file->spew( iomode => '>:utf8', $contents );
```

Comment:
It's probably worth dying if the `$extra_title_page` exists as well as the `$custom_title_page`.

Reply:
Fixed in fe0d7f739080b267866ef3519201e6fd893daaf2